### ISSUE TITLE
Add serde support

### ISSUE THREAD

**Issue by connorkuehl** [enarx-archive/sev#28]
The AMD SEV types that are relevant for attestation should support serde so that we can pass these structures as CBOR-encoded messages.

I've already done the brunt of this work as part of my work on the remote AMD SEV attestation, but I'm opening this issue to track that work.

Depends on #27

**Comment by connorkuehl** [comment]
Closed in https://github.com/enarx/sev/pull/33
